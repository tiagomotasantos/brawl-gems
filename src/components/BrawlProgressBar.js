import React from 'react';
import Color from 'color';

const BrawlProgressBar = ({ value, icon, current, max, color }) => (
    <div className="progress-bar">
        <div 
            className="progress-bar__inner"
            style={{
                backgroundColor: color,
                borderBottom: '1.8rem solid ' + Color(color).darken(0.4),
                width: current * 100 / max
            }}
        >
        </div>
        <div className="progess-bar__values-container">
            <span className="progess-bar__values">{current}/{max}</span>
            <img alt="Adfsa" src={icon} className="progess-bar__icon" />
            <span className="progress-bar__value">{value}</span>
        </div>
    </div>
);

export default BrawlProgressBar;