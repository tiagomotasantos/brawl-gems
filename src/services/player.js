
import axios from './axios';

export const getPlayer = (tag) => {
    return axios.get(`player?tag=${tag}`);
}