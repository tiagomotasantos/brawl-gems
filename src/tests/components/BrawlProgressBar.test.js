import React from 'react';
import { shallow } from 'enzyme';
import BrawlProgressBar from '../../components/BrawlProgressBar';
import player from '../fixtures/player';

test('should render BrawlProgressBar', () => {
    const wrapper = shallow(<BrawlProgressBar 
        value={player.expLevel} 
        icon="/images/level.png" 
        current={100} max={10000} color="#3EBEED"
    />);
    expect(wrapper).toMatchSnapshot();
});