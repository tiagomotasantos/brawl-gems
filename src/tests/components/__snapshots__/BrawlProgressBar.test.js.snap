// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BrawlProgressBar 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__inner"
    style={
      Object {
        "backgroundColor": "#3EBEED",
        "borderBottom": "1.8rem solid hsl(196.10000000000002, 82.9%, 35.2%)",
        "width": 1,
      }
    }
  />
  <div
    className="progess-bar__values-container"
  >
    <span
      className="progess-bar__values"
    >
      100
      /
      10000
    </span>
    <img
      alt="icon"
      className="progess-bar__icon"
      src="/images/level.png"
    />
    <span
      className="progress-bar__value"
    >
      59
    </span>
  </div>
</div>
`;
